default-fields/0000-default-fields.md

```diff
@@ -6,31 +6,206 @@
 # Summary
 [summary]: #summary
 
-One para explanation of the feature.
+Allow struct definitions to supply default values for individual fields,
+and then allow those fields to be omitted from struct initialisation:
+
+```rust
+struct Foo {
+    a: &'static str,
+    b: bool = true,
+    c: i32,
+    d: Vec<i32> = Vec::default()
```

Comment:
Hmm, so we'd have a lang item that looks a little something like (pseudo):

```rust
struct DefaultField(Ty, DefaultFieldKind)

enum DefaultFieldKind {
    Const(Expr),
    Default
}
```

Where default would expand to `Default::default()`?